panther_analysis_tool/main.py

```diff
@@ -420,6 +422,19 @@ def classify_analysis(
     return (global_analysis, analysis, invalid_specs)
 
 
+def handle_wrong_key_error(err: SchemaWrongKeyError, keys: list) -> Exception:
+    regex = r"Wrong key(?:s)? (.+?) in (.*)$"
+    matches = re.match(regex, str(err))
+    msg = '{} not in list of valid keys: {}'
+    try:
+        if matches:
+            raise SchemaWrongKeyError(msg.format(matches.group(1),
+                                                 keys)) from err
+        raise SchemaWrongKeyError(msg.format('UNKNOWN_KEY', keys)) from err
+    except SchemaWrongKeyError as exc:
+        return exc
+
+
```

Comment:
I didn’t see a straightforward way to handle re-writing the error output for `SchemaWrongKeyError`.  `schema` does allow for “user friendly error reporting” reporting (https://pypi.org/project/schema/), however I believe we would have to specify the error at the `Schema` level.  This means that even non-SchemaWrongKeyError exceptions would have our customized error message (for example: `SchemaUnexpectedTypeError` would share the same message at `SchemaWrongKeyError`).  This workaround keeps the original `SchemaWrongKeyError` put creates a new error message based on the the unknown key that was identified and keys for the schema being consideration.  Let me know what you think, I’m not 100% sure about this since it seems fragile (regex matching output that could potentially change)

Reply:
Looking at the source code: https://github.com/keleshev/schema/blob/master/schema.py#L32-L38, you can see there are `.autos` and `.errors` attributes in the base `SchemaErrors` class:

```python
class SchemaError(Exception):
    """Error during Schema validation."""

    def __init__(self, autos, errors=None):
        self.autos = autos if type(autos) is list else [autos]
        self.errors = errors if type(errors) is list else [errors]
        Exception.__init__(self, self.code)
```

You may be able to use those to get closer to the information you want. For example:

```
In [10]: try:
    ...:     TYPE_SCHEMA.validate({'goodbye': 'world'})
    ...: except SchemaError as ex:
    ...:     print(ex.autos)
    ...:     print(ex.errors)
    ...:
["Missing key: 'AnalysisType'"]
[None]
```

But ultimately, it looks like those will still just be a list of error messages. There's only so much you can do to compensate for a library with immature error handling; they should have error classes with the information attached. But since they don't, parsing the msg seems like a reasonable workaround.

Reply:
Yea, the autos attr contains the original error message we are seeing (the large output we are trying to address - "Wrong key X in ..." ) I'll keep this as a regex extraction for now based on both of your input.